src/recon.rs

```diff
@@ -1013,34 +1084,59 @@ fn decode_coefs<BD: BitDepth>(
         let cf = &mut cf;
         (rc, dc_tok) = match tx_class {
             TxClass::TwoD => decode_coefs_class::<{ TxClass::TwoD as _ }, BD>(
-                ts_c, t_dim, chroma, scratch, eob, tx, dbg, cf,
+                ts_c,
+                &tdim!(),
```

Comment:
Should this be passed as a const generic, too?

Reply:
We can, and it did seem to work equally. I can push this change here if you'd like. I'm trying to avoid pushing too many changes to not incur many more rounds of measurements to everybody.